pkg/odo/cli/delete/component/component.go

```diff
@@ -85,6 +90,18 @@ func (o *ComponentOptions) Complete(ctx context.Context, cmdline cmdline.Cmdline
 	} else {
 		o.namespace = o.clientset.KubernetesClient.GetCurrentNamespace()
 	}
+
+	// Limit access to platforms if necessary
+	if !feature.IsEnabled(ctx, feature.GenericRunOnFlag) {
+		o.clientset.PodmanClient = nil
+	}
+	switch fcontext.GetRunOn(ctx, "") {
+	case commonflags.RunOnCluster:
+		o.clientset.PodmanClient = nil
+	case commonflags.RunOnPodman:
+		o.clientset.KubernetesClient = nil
+	}
+
```

Comment:
I think it should go on L79, because if `o.name == "" ` is satisfied, it will return early.